src/core/Class.js

```diff
@@ -17,6 +17,8 @@ Class.extend = function (props) {
 	// Returns a Javascript function that is a class constructor (to be called with `new`).
 	var NewClass = function () {
 
+		Util.setOptions(this, {});

```

Comment:
```suggestion
		Util.setOptions(this);
```

Reply:
And even more: as now it is guaranteed that every object has distinct `options` we could refactor the code, adding this to constructor function:
```js
		this.options = this.options ? Util.create(this.options) : {};
```
And removing this:
https://github.com/Leaflet/Leaflet/blob/620bc3e9c5d282090f429ac21d28d00fbffdb3db/src/core/Util.js#L133-L135

Reply:
Where do you want me to add the code `this.options = this.options ? Util.create(this.options) : {};`? Instead of `Util.setOptions(this):`?

Reply:
> Where do you want me to add the code

I suppose this PR is finished, and we will wait for other maintainers to review it.
And if it will be merged - then we will continue from that part.